### DISCUSSION TITLE
Steam not running off of a ZFS pool, to be expected?

### DISCUSSION THREAD

**Discussion by Mokerak** [openzfsonwindows/openzfs#331] [Q&A]
I'm just experimenting right now, I'm new to ZFS and have very little experience with Linux/UNIX. Should it be too much of me to expect steam to be able to run off a ZFS drive.

I get the error SteamUI.DLL missing every time I run it off the ZFS D: drive, but when I run it on my C: drive it runs great. I do indeed have SteamUI.dll this time since I just copied and pasted my working steam over to the D drive this time, before I tried installing directly to D. I still get the same error. I think this is just the error steam says no matter what when something is wrong with the install.

**Comment by EchterAgo** [comment]
![image](https://github.com/openzfsonwindows/openzfs/assets/922233/861d32df-c7ae-4302-a934-8d7a1918bb71)

Seems to work fine here. You aren't still running RC5, are you?

Also, even though Steam starts here, there have been other problem reports, see #221.

What dataset options did you set? For Windows datasets you should use `-O casesensitivity=insensitive -O normalization=formD` because many Windows programs don't expect a case sensitive file system.

**Reply by EchterAgo** [reply]
![image](https://github.com/openzfsonwindows/openzfs/assets/922233/8e2f3d16-0404-4a35-aab0-c6ec059d17f6)

**Reply by EchterAgo** [reply]
Oh no, I just realized I wasted all my progress towards the game's "Go Outside" achievement...

**Reply by Mokerak** [reply]
I used all the options that a guide suggested

![image](https://github.com/openzfsonwindows/openzfs/assets/34557152/c21b374e-ce2d-4b52-97a3-683eda6c9970)

**Reply by Mokerak** [reply]
I've also had issues with certain games not running on it but some do. Like Europa Universalis 4 runs on it, but Stellaris wont. Another thing, running a Minecraft server on ZFS makes an issue about Fabric not being installed, but on other drives its fine

**Reply by Mokerak** [reply]
I am running 5, is there an easy way to update it?

**Reply by Mokerak** [reply]
Also is there a way I can double check what options I used?

**Reply by Mokerak** [reply]
Well running the server on rc6 didn't work

![20231112_214852](https://github.com/openzfsonwindows/openzfs/assets/34557152/407de96c-1b5f-4936-b8e6-7c14ba7067ba)

**Reply by EchterAgo** [reply]
Is there a problem with just installing the update?

You can check using the `zpool history` command.

**Reply by Mokerak** [reply]
It blue screens whenever I try running the server, lemme see if it bluescreens just from anything using it in general

**Reply by EchterAgo** [reply]
We'd need the dump file analyzed to see what the issue is, the BSOD screen has only very little information.

https://openzfsonosx.org/wiki/Windows_BSOD

**Reply by Mokerak** [reply]
Should I upload it or something? 1.7GB is the size of mine

25MB is the limit for discussion posts

**Reply by EchterAgo** [reply]
When you say "running the server" does that mean early in boot or do you have some automatic zpool import active?

**Reply by Mokerak** [reply]
I'm just running a bat file on the zpool drive, whenever it gets to a certain point while running commands to load mods and stuff it just crashes my whole computer. I ran some other games and they didn't do this so updating the ZFS to version 6 didn't brick anything I think. 

EDIT: So not during boot at all or anything automatic. I'm just manually running the file after the ZFS pool is imported after launch

**Reply by Mokerak** [reply]
Its running a fabric server I think. Sorry I'm not very proficient in Minecraft stuff. 

![Capture](https://github.com/openzfsonwindows/openzfs/assets/34557152/92f41a31-3fe0-4c24-9c62-0255ae2fe2c5)

**Reply by EchterAgo** [reply]
Oh, I assumed you meant your host is a server and you get that at startup.

The easier you make it for someone to investigate your issue the more likely you are to get help. I'd start with creating an issue describing the problem in detail and including steps how to produce it, for example how to start a minimal Minecraft server that produces the issue. Also try to include the information you get from following the steps in https://openzfsonosx.org/wiki/Windows_BSOD. You can of course also upload the memory dump somewhere, that can help.

**Comment by EchterAgo** [comment]
What I noticed that doesn't work is the Desktop icons Steam creates. It seems to create URL files on the Desktop (NTFS), but their icons are blank and the size on disk is 0 bytes?

![image](https://github.com/openzfsonwindows/openzfs/assets/922233/a7960902-0ee2-4c9b-9a18-d1a93d68229f)

Not sure this is actually related to ZFS. I'm not sure, but I recall seeing something like this without ZFS too.

**Reply by EchterAgo** [reply]
They also don't start when I double click them, asking me to choose a program instead. The contents of the file are:

```ini
[{000214A0-0000-0000-C000-000000000046}]
Prop3=19,0
[InternetShortcut]
IDList=
IconIndex=0
URL=steam://rungameid/3620
IconFile=D:\Steam\steam\games\4ac07602eb3f1887c880f95b469b99cfa29a8b21.ico
```

**Reply by lundman** [reply]
That looks more like it should live in an EA?

**Reply by EchterAgo** [reply]
No, that actually is just a normal URL file, also it is created on NTFS like this, only the `IconFile` is referencing a path on ZFS. I normally would expect this to work like it is, maybe something wrong on this system.

**Reply by EchterAgo** [reply]
After opening the url file Windows asks which application to open it with. When I select "Internet browser" (the default in the list) then Steam starts as expected and the URL file doesn't display its `.url` extension anymore. For some reason the url file association has gone missing on this system? The icon still does not display, but that might be a matter of clearing the shell icon cache. I also tried running a game and it works.

**Reply by EchterAgo** [reply]
Yea, after clearing the icon cache the icons display now too. This is very likely a misconfiguration on this system, although I don't remember making it.

```batch
@echo off

echo "Clearing the icon cache"

taskkill /IM "explorer.exe" /F
cd /d "%LOCALAPPDATA%"\Microsoft\Windows\Explorer 
attrib –h iconcache_*.db 
del iconcache_*.db 
start explorer
```
